Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -159,10 +161,11 @@ namespace ROS2
         AZStd::string instanceName = AZStd::string::format("%s_%d", spawnableName.c_str(), m_counter++);
         for (AZ::Entity* entity : view)
         { // Update name for the first entity with ROS2Frame in hierarchy (left to right)
-            const auto* frameComponent = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(entity);
+            auto* frameComponent = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(entity);
             if (frameComponent)
             {
                 entity->SetName(instanceName);
+                frameComponent->UpdateNamespaceConfiguration(spawnableNamespace, NamespaceConfiguration::NamespaceStrategy::Custom);
```

Comment:
Have you considered checking if spawnableNamespace is empty? As I understand it will be empty if omitted by a service caller, perhaps in such cases namespace shall not be altered?